main.tf

```diff
@@ -83,15 +90,16 @@ resource "aws_lb" "default" {
 }
 
 module "default_target_group_label" {
-  source     = "cloudposse/label/null"
-  version    = "0.24.1"
-  attributes = concat(module.this.attributes, ["default"])
-  context    = module.this.context
+  source          = "cloudposse/label/null"
+  version         = "0.24.1"
+  attributes      = concat(module.this.attributes, ["default"])
+  id_length_limit = var.target_group_name_max_length
+  context         = module.this.context
 }
 
 resource "aws_lb_target_group" "default" {
   count                = module.this.enabled ? 1 : 0
-  name                 = var.target_group_name == "" ? module.default_target_group_label.id : var.target_group_name
+  name                 = var.target_group_name == "" ? module.default_target_group_label.id : substr(var.target_group_name, 0, var.target_group_name_max_length)
```

Comment:
why you need to `substr` if then the length is set the label.id will be of the required length?

Reply:
The substr is on the target group name which is used to override the label id

Reply:
got it thanks